internal/blueprint/internal/scheme/transform.go

```diff
@@ -0,0 +1,315 @@
+		obj = toV2(obj)
+	}
+
+	switch kind := getString(obj, "kind"); kind {
+	case "Project":
+		return toInternalProject(obj)
+	case "Service":
+		return toInternalService(obj)
+	case "Environment":
+		return toInternalEnvironment(obj)
+	case "Builder", "Deployer", "Pusher", "Tagger":
+		return toInternalExecutor(obj)
+	default:
+		panic(fmt.Errorf("unsupported kind: %s", kind))
+	}
+}
+
+func toInternalProject(obj any) any {
+	files := []any{}
+
+	for _, f := range getSlice(obj, "files") {
+		if isString(f) {
+			files = append(files, map[string]any{"glob": f})
+		} else {
+			if has(f, "git") {
+				gitFiles := get(f, "git", "files")
+				if isString(gitFiles) {
+					gitFiles = []any{gitFiles}
+				}
+				for _, glob := range getSlice(gitFiles) {
+					files = append(files, map[string]any{
+						"glob": glob,
+						"git": map[string]any{
+							"url": getString(f, "git", "url"),
+							"rev": getString(f, "git", "rev"),
+						},
+					})
+				}
+			}
+		}
+	}
+
+	return map[string]any{
+		"kind":      "Project",
+		"files":     files,
+		"name":      getString(obj, "name"),
+		"variables": getMap(obj, "variables"),
+	}
+}
+
+func toInternalService(obj any) any {
+	toBuild := mapSlice(getSlice(obj, "artifacts"), toInternalEntry)
+	tags := mapSlice(getSlice(obj, "tags"), toInternalEntry)
+	releases := mapSlice(getSlice(obj, "releases"), toInternalEntry)
+
+	toPush := []any{}
+	for i, v := range getSlice(obj, "artifacts") {
+		e := toInternalEntry(i, or(get(v, "push"), v))
+		if getString(e, "type") != "" {
+			toPush = append(toPush, e)
+		}
+	}
+
+	tasks := map[string]any{}
+	for k, v := range getMap(obj, "tasks") {
+		tasks[k] = mapSlice(getSlice(v), toInternalEntry)
+	}
+
+	return map[string]any{
+		"kind": "Service",
+		"name": getString(obj, "name"),
+		"build": map[string]any{
+			"artifacts": map[string]any{
+				"toBuild": toBuild,
+				"toPush":  toPush,
+			},
+			"tags": tags,
+		},
+		"deploy": map[string]any{
+			"releases": releases,
+		},
+		"run": map[string]any{
+			"tasks": tasks,
+		},
+	}
+}
+
+func toInternalEnvironment(obj any) any {
+	return map[string]any{
+		"kind":           "Environment",
+		"name":           getString(obj, "name"),
+		"deployServices": getSlice(obj, "deployServices"),
+		"variables":      getMap(obj, "variables"),
+	}
+}
+
+func toInternalExecutor(obj any) any {
+	schema, err := json.Marshal(getMap(obj, "schema"))
+	if err != nil {
+		panic(err)
+	}
+	return map[string]any{
+		"kind":   getString(obj, "kind"),
+		"name":   getString(obj, "name"),
+		"js":     getString(obj, "js"),
+		"schema": string(schema),
+	}
+}
+
+func toInternalEntry(i int, obj any) any {
+	if obj == false {
+		return nil
+	}
+	if isString(obj) {
+		return map[string]any{"index": int64(i), "type": obj, "spec": nil}
+	}
+	for k, v := range getMap(obj) {
+		if k != "push" {
+			return map[string]any{"index": int64(i), "type": k, "spec": v}
+		}
+	}
+	return nil
+}
+
+func toV2(obj any) any {
```

Comment:
V2 of what? Schema package seems to do everything. Mapping file, doing migration, etc. I would split it into packages to make it more legible.